### PR TITLE
Fix FileDialog file name autocompletion with filters

### LINKED ISSUE #38195: Incorrect File Autocompletion When Using FileDialog Filters
**Godot version:** 3.2.1.stable.official

**OS/device including version:** Windows 10 Home Version 1903

**Issue description:** When the `filters` property for FileDialog nodes is set, weird autocompletion occurs when the dialog is in use. If you click the dropdown menu near the bottom right corner of the dialog, and select a specific file type (eg. the filter `*.png ; PNG Files`), the file LineEdit is autocompleted incorrectly (in this example, it would be ".*.png ; png files").

If you do this multiple times (for example, switching between file types), the file extension will continue to be appended at the end; for some weird reason, the semicolon and lowercase description only ever appears once at the end and gets overwritten if used multiple times. See below for an image showing the result.

image

**Steps to reproduce:**

1. Create a scene with a FileDialog. In my code, I made it popup when a TextureRect is pressed, and the bug showcase project uses a Button, but any way should work.
2. Set the `filters` property. Define at least two file types. I used `*.png ; PNG Files` and `*.jpg ; JPG Files`.
3. Run the program and open the FileDialog.
4. Select one of the two custom filters by selecting the dropdown menu near the bottom right corner. Observe the weird autocompletion.
5. Do this a few more times. Observe the additional autocompletion.

**Minimal reproduction project:** [FileDialog Bug Showcase.zip](https://github.com/godotengine/godot/files/4532269/FileDialog.Bug.Showcase.zip)

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #38195

### CODE REVIEW

**Comment by dominiks** [comment]
Any news on merging this and a backport into 3.2? Would love to have this fix in my project.

File: scene/gui/file_dialog.cpp [hunk 1]
```diff
@@ -414,7 +414,12 @@ void FileDialog::update_file_name() {
 		String filter_str = filters[idx];
 		String file_str = file->get_text();
 		String base_name = file_str.get_basename();
-		file_str = base_name + "." + filter_str.strip_edges().to_lower();
+		Vector<String> filter_substr = filter_str.split(";");
+		if (filter_substr.size() >= 2) {
+			file_str = base_name + "." + filter_substr[0].strip_edges().lstrip("*.").to_lower();
```
Comment (akien-mga) [inline scene/gui/file_dialog.cpp:419]: Why use `lstrip` and not `get_extension` in this case, like for the `== 1` case?

Comment (ThakeeNathees) [reply scene/gui/file_dialog.cpp:419]: I saw it from `EditorFileDialog::update_file_name`, not sure why though. should I change it?
https://github.com/godotengine/godot/blob/96884ef41ecbadd8e76f5e42af5f5cd53a902de0/editor/editor_file_dialog.cpp#L675
@akien-mga

Comment (akien-mga) [reply scene/gui/file_dialog.cpp:419]: Ah that would have been useful to mention from the start, it's easier to review a PR when we know it's tested code from another similar class :)

I think both are equivalent and should give the same output, so I'm not sure with EditorFileDialog should use two different calls. Written by two different contributors it seems and this passed through review: 319840bad1754ce33ec60000df7972e73401860d.

It seems I had actually called it out as not being perfect in https://github.com/godotengine/godot/pull/36699#issuecomment-594417487, but didn't think to suggest just using `get_extension()`.

I'd say you can use `get_extension()` here and in `EditorFileDialog` (and check that it still works as expected of course).

Comment (akien-mga) [reply scene/gui/file_dialog.cpp]: Pushed that change myself to unblock that PR two years later ;)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Rebased and tested on current `master`, it does fix the bug.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**